src/main/java/com/sugarcrm/candybean/candybeanRunner/CandybeanRunner.java

```diff
@@ -0,0 +1,332 @@
+ * how to extend JUnit runners, I've attempted to keep the documentation to a level that
+ * anyone can figure out what's going on without reading the docs (even though you should
+ * anyways).
+ */
+public class CandybeanRunner extends BlockJUnit4ClassRunner {
+
+	protected final Candybean candybean = Candybean.getInstance();
+	protected final int retryCount = Integer.parseInt(candybean.config.getValue("runner.retryCount", "0"));
+	protected final int timeout = Integer.parseInt(candybean.config.getValue("runner.timeout", "0"));
+	protected final int cleanupTimeout = Integer.parseInt(
+			candybean.config.getValue("runner.cleanupTimeout", String.valueOf(timeout)));
+	protected final boolean rerunIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.rerunIfTimedOut", "false"));
+	protected final boolean runBeforeIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.runBeforeIfTimedOut", "true"));
+	protected final boolean runAfterIfTimedOut = Boolean.parseBoolean(candybean.config.getValue("runner.runAfterIfTimedOut", "true"));
+	final boolean INTERRUPT = true;
+
+	/**
+	 * Creates a BlockJUnit4ClassRunner to run klass
+	 *
+	 * @param klass The class to initialize
+	 * @throws InitializationError
+	 */
+	public CandybeanRunner(Class<?> klass) throws InitializationError, CandybeanException{
+		super(klass);
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void run(final RunNotifier notifier) {
+		/* Construct statement that when evaluated:
+		 *  Runs @BeforeClass
+		 *  Runs All children tests with runChild
+		 *  Runs @AfterClass
+		 *
+		 * If the test fails, it notifies the test runner
+		 *
+		 * This method is called on each test class. Evaluating
+		 * the statement then runs runChild on each test within
+		 * the class.
+		 */
+		final Statement statement = classBlock(notifier);
+		try {
+			statement.evaluate();
+		} catch (StoppedByUserException e) {
+			throw e;
+		} catch (Throwable e) {
+			notifier.fireTestFailure(new Failure(getDescription(), e));
+		}
+	}
+
+	/**
+	 * Runs each test using retries and timeouts. The general idea of the this
+	 * method is that it constructs a Callable that when run, runs the test. It
+	 * attempts to retrieve the results of the callable after a certain timeout.
+	 * If the test wasn't ready, or the test failed, it reruns the callable if
+	 * applicable up to some specified number of times.
+	 *
+	 * @param method The test method to run
+	 * @param notifier The test notifier to update
+	 */
+	@Override
+	protected void runChild(final FrameworkMethod method, RunNotifier notifier) {
+		final Description description = describeChild(method);
+
+		// Check the annotations, it the test is marked ignore, ignore it.
+		// If you wanted add a check for custom annotations, perhaps you
+		// wanted an @DoNotReRun annotation, you would check for it here
+		if (method.getAnnotation(Ignore.class) != null) {
```

Comment:
Should do, I believe the logic for that is handled in Grimoire, the runner shouldn't change how it's handled. I don't however, currently have the ability to check that.